### DISCUSSION TITLE
🚀 Distro 2022.02.0 Released

### DISCUSSION THREAD

**Discussion by Nicogene** [robotology/community#571] [Releases]
# 2022.02.0 Distro Announcement

## Brief description of the announcement

Hello @robotology/everyone  👋

The [iCub project](https://icub.iit.it/) is happy to announce our **Software Distro 2022.02.0** 📦

This is a major release that comprises **binary packages** as well. Therefore, follow the [instructions][1] available in our documentation to upgrade your system and/or download the new dependencies.

You can find the tags composing this distro in our [Software Versioning Table][2].

The apt binaries cover the following platforms:
- Ubuntu 20.04 Focal 64-bits.
- Ubuntu 18.04 Bionic 64-bits.
- Debian 10 Buster 64-bits.

The conda binaries cover the following platforms:
- Linux x64 64-bits.
- Windows 10 x64 64-bits.
- macOS x64 64-bits.

## Detailed context

This distro is shipped with several updates and fixes, including the ones listed hereafter:

#### 🔘 Upgraded some specific robots to the use of the latest NWS/NWC layers
For details, see:
- https://github.com/robotology/robots-configuration/pull/321

#### 🔘 Fixed the encoder consistency test
The coupling matrices are now retrieved online directly from the robot avoiding possible mismatches among the content of the robot configuration files and the test initialization files.
- The problem was reported in https://github.com/robotology/robots-configuration/issues/252.
- The fix landed in https://github.com/robotology/icub-tests/pull/60.

#### 🔘 `blenderController` robot-bazaar application
The [`blenderCotroller`](https://robot-bazaar.iit.it/applications/29) application has been added to [robot-bazaar](https://robot-bazaar.iit.it) 🚀

It is now possible to easily deploy the [`blenderRCBPanel`](https://github.com/robotology/blender-robotics-utils#blenderrcbpanel-) using the [RAPID deployment system](https://robot-bazaar.iit.it/faq), you can control/animate your iCub through Blender in few clicks. ✨

<img width="355" alt="image" src="https://user-images.githubusercontent.com/19152494/151566373-7be71433-633f-4b59-b79b-1b2ccf4335bd.png">

[1]: https://icub-tech-iit.github.io/documentation/sw_installation/
[2]: https://icub-tech-iit.github.io/documentation/sw_versioning_table/#table

**Comment by traversaro** [comment]
### Extra Updates 

#### 🔘 robotology-distro and robotology-distro-all conda metapackages

As an experimental feature, from the 2022.02.0 distro we provide also the `robotology-distro` conda metapackage, that simplifies the process of installing the packages that you want to install at the version specified by a given distro.

With  `robotology-distro`, if somebody wanted for example to install `yarp`, `icub-main`, `gazebo-yarp-plugins` and `icub-models` corresponding to the 2022.02.0 distro, they could do it easily by doing
~~~
mamba install -c conda-forge -c robotology robotology-distro=2022.02.0 yarp icub-main  gazebo-yarp-plugins icub-models
~~~

Note that in this case the `robotology-distro` package is empty, and so the user needs anyhow to specify the packages that needs to be installed. If instead a user wants to install **all the packages** that are part of a robotology distro, the `robotology-distro-all` package can be installed:
~~~
mamba install -c conda-forge -c robotology robotology-distro-all=2022.02.0
~~~

Until the 2022.02.0 distro, what was suggested is that the user need to manually inspect the tables in https://icub-tech-iit.github.io/documentation/sw_versioning_table/ to extract the version for each package that they need, but we hope that the new feature is more user friendly. 

**As this is an experimental feature, it is for now only available on Windows and Linux, and not on macOS.**
